cli/state.rs

```diff
@@ -250,10 +250,10 @@ impl ThreadSafeState {
         msg::MediaType::Json => {
           state_.json_compiler.compile_async(state_.clone(), &out)
         }
-        msg::MediaType::TypeScript => {
+        msg::MediaType::TypeScript | msg::MediaType::TSX => {
           state_.ts_compiler.compile_async(state_.clone(), &out)
         }
-        msg::MediaType::JavaScript => {
+        msg::MediaType::JavaScript | msg::MediaType::JSX => {
```

Comment:
Are you sure about this.  This will leave the `.jsx` untouched if the `checkJs` compiler flag is not set.  I can't see that being desirable.

We should have tests around `.jsx` and we should have an expected behaviour of how they are handled.  We also need it consistent with how the compiler behaves, where by default we have allow JS but not check JS, so how does that effect the emit?

Reply:
Oh, I absolutely forgot it. Hmm... should we compile `.jsx` files as TSX?

Reply:
Well, not compile as TSX, more that we will _always_ attempt to transpile them, move it to the previous match clause, but I would check that that works as expected, when the `checkJs` compiler flag is not enabled (which is the default), which the integration test should prove.